File: dani-eth/frontend/src/types/campaign.ts
export type CampaignStatus = 'started' | 'running' | 'paused' | 'stopped' | 'finished';
export type FindingStatus = 'pending' | 'reviewed' | 'false_positive';
export type FindingSeverity = 'critical' | 'high' | 'medium' | 'low' | 'info';

export interface Finding {
  finding_id: string;
  type: string;
  severity: FindingSeverity;
  description: string;
  evidence: string;
  status: FindingStatus;
  cve: string | null;
  port?: number;
  service?: string;
}

export interface CampaignStatusResponse {
  campaign_id: string;
  status: CampaignStatus;
  phase: string | null;
  progress: number | null;
  findings: Finding[];
  logs: string[];
  started_at: string | null;
  finished_at: string | null;
}

export interface CampaignStart {
  target_id: string;
  target: string;
  scan_type: 'full' | 'quick' | 'custom';
  scope: string[];
}

export interface RemediationItem {
  finding_id: string;
  type: string;
  severity: FindingSeverity;
  action: string;
  reference: string;
  remediated: boolean;
}

export interface RemediationPlan {
  campaign_id: string;
  plan: RemediationItem[];
}

export interface FindingStatusUpdate {
  status: FindingStatus;
}

export interface FindingRemediatedUpdate {
  remediated: boolean;
  notes?: string;
}

File: dani-eth/backend/app/schemas/campaign.py
from pydantic import BaseModel
from typing import Literal


class CampaignStart(BaseModel):
    target_id: str
    target: str
    scan_type: Literal["full", "quick", "custom"] = "quick"
    scope: list[str] = ["ports", "web"]


class CampaignStatusResponse(BaseModel):
    campaign_id: str
    status: Literal["started", "running", "paused", "stopped", "finished"]
    phase: str | None = None
    progress: int | None = None
    findings: list[dict] = []
    logs: list[str] = []
    started_at: str | None = None
    finished_at: str | None = None


class FindingStatusUpdate(BaseModel):
    status: Literal["pending", "reviewed", "false_positive"]


class FindingRemediatedUpdate(BaseModel):
    remediated: bool
    notes: str | None = None